### PR TITLE
logpipe-source: disable multi-line-timeout

### PR DESCRIPTION
**Pull request by vivinperis**
I have taken care of the earlier changes. Sorry for the issues that inadvertently popped up

Fixes: https://github.com/syslog-ng/syslog-ng/issues/3178

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by mehul-m-prajapati** [comment]
Hi @vivinperis,

You need to read this first (https://github.com/syslog-ng/syslog-ng/blob/master/CONTRIBUTING.md#commits)

There should be only one commit which resolves the issue.

**Comment by vivinperis** [comment]
I figured out my mistake later. I'm not able to delete my old commits and merge them into one

**Comment by mehul-m-prajapati** [comment]
> I figured out my mistake later. I'm not able to delete my old commits and merge them into one

You can squash commits https://github.com/wprig/wprig/wiki/How-to-squash-commits

File: modules/affile/affile-grammar.ym [hunk 1]
```diff
@@ -257,12 +256,34 @@ multi_line_option
...
+	    	free($3);
 	  }
 	| KW_MULTI_LINE_TIMEOUT '(' nonnegative_integer ')'
 	  {
-	    file_reader_options_set_multi_line_timeout(last_file_reader_options, ($3 * 1000));
+	    	file_reader_options_set_multi_line_timeout(last_file_reader_options, ($3 * 1000));
+	  }
+	;
+
+multi_line_option_change
```
Comment (furiel) [inline modules/affile/affile-grammar.ym]: This solution can work, but you are duplicating the non-timeout related options, which is harder to maintain.

Instead, please partition the `multi_line_option`. `multi_line_option = multi_line_option_without_timeout + timeout`. Then you can `use multi_line_option_timeout` in the logpipe source.

Comment (vivinperis) [reply modules/affile/affile-grammar.ym]: fixes: https://github.com/syslog-ng/syslog-ng/issues/3178
Made the necessary changes to the code

**Comment by furiel** [comment]
@kira-syslogng ok to test

File: modules/affile/affile-grammar.ym (hunk 1 above)
Comment (furiel) [reply modules/affile/affile-grammar.ym]: Thanks!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by vivinperis** [comment]
Why is continuous integration giving build error?

**Comment by furiel** [comment]
I am fine with this implementation. I tried it out, and it works. Please squash the commits. Then this is ready to be merged from my side.

**Comment by furiel** [comment]
> Why is continuous integration giving build error?

We experience some instability with the cmake-arm build. But it is not caused by your change.

The commit check enforces the commit formats according to this: https://github.com/syslog-ng/syslog-ng/blob/master/CONTRIBUTING.md#commits. After squashing and rewording your commit according to that, the test should pass.

**Comment by vivinperis** [comment]
I managed to squash all the commits.
Please check and let me know if its fine now

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by furiel** [comment]
This last push added some unwanted lines from half-resolved conflicts. Also, the number of commits further increased from 9 to 11.

If you got stuck, feel free to PM me on gitter (username `@furiel`), and I help you with that. This is more like about the steep learning curve of git, than the actual code change.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**